ironfish/src/network/peerNetwork.ts

```diff
@@ -710,6 +714,8 @@ export class PeerNetwork {
       return false
     }
 
+    this.onTransactionReceived.emit(transaction, received)
```

Comment:
What would happen if someone attached a slow event handler? Should we have this after the mem pool processes it?

Reply:
I thought handlers execute async? Or does it have some affect on the timing?

Reply:
Javascript is single threaded, so what ever happens in that handler will take over the event loop until all handlers are finished.